chia/rpc/wallet_rpc_api.py

```diff
@@ -1512,7 +1517,8 @@ async def nft_add_uri(self, request) -> Dict:
         try:
             uri = request["uri"]
             key = request["key"]
-            nft_coin_info = nft_wallet.get_nft_coin_by_id(bytes32.from_hexstr(request["nft_coin_id"]))
+            nft_coin_id = decode_puzzle_hash(request["nft_coin_id"])
```

Comment:
As above, I would recommend treating this value as being either a bech32m string or a hex string. And if that's too complicated, I think a hex string is the better option.